src/plugins/custom_integrations/common/index.ts

```diff
@@ -125,3 +126,146 @@ export interface CustomIntegration {
 
 export const ROUTES_APPEND_CUSTOM_INTEGRATIONS = `/internal/${PLUGIN_ID}/appendCustomIntegrations`;
 export const ROUTES_REPLACEMENT_CUSTOM_INTEGRATIONS = `/internal/${PLUGIN_ID}/replacementCustomIntegrations`;
+
+/*
+ * Definitions for the language integrations
+ */
+const ELASTIC_WEBSITE_URL = 'https://www.elastic.co';
+const ELASTICSEARCH_CLIENT_URL = `${ELASTIC_WEBSITE_URL}/guide/en/elasticsearch/client`;
+
+/*
+  - exportLanguageUiComponent: controls whether the integration should export a UI component for language clients to Fleet UI;
+    if false, the URL falls back to `docUrlTemplate`; this logic is handled in `registerCustomIntegration`
+  - integrationsAppUrl: url in Integrations App where the components defined under `public/components/fleet_integration` will be rendered
+      only exported when `exportLanguageUiComponent` is true
+*/
+export interface LanguageIntegration {
+  id: string;
+  title: string;
+  icon?: string;
+  euiIconName?: string;
+  description: string;
+  docUrlTemplate: string;
+  integrationsAppUrl: string;
+  exportLanguageUiComponent: boolean;
+}
+
+export const languageIntegrations: LanguageIntegration[] = [
```

Comment:
nit: the language integration related code could move to a separate file, e.g. `language_integrations.ts`